sdk/core/azure-core/src/main/java/com/azure/core/util/CoreUtils.java

```diff
@@ -376,4 +379,45 @@ public static Context mergeContexts(Context into, Context from) {
 
         return returnContext;
     }
+
+    /**
+     * Writes a {@link ByteBuffer} into an {@link OutputStream}.
+     * <p>
+     * This method provides writing optimization based on the type of {@link ByteBuffer} and {@link OutputStream}
+     * passed. For example, if the {@link ByteBuffer} has a backing {@code byte[]} this method will access that directly
+     * to write to the {@code stream} instead of buffering the contents of the {@link ByteBuffer} into a temporary
+     * buffer.
+     *
+     * @param buffer The {@link ByteBuffer} to write into the {@code stream}.
+     * @param stream The {@link OutputStream} where the {@code buffer} will be written.
+     * @throws IOException If an I/O occurs while writing the {@code buffer} into the {@code stream}.
+     */
+    public static void writeByteBufferToStream(ByteBuffer buffer, OutputStream stream) throws IOException {
+        // First check if the buffer has a backing byte[]. The backing byte[] can be accessed directly and written
+        // without an additional buffering byte[].
+        if (buffer.hasArray()) {
+            // Write the byte[] from the current view position to the length remaining in the view.
+            stream.write(buffer.array(), buffer.position(), buffer.remaining());
+
+            // Update the position of the ByteBuffer to treat this the same as getting from the buffer.
+            buffer.position(buffer.position() + buffer.remaining());
```

Comment:
Is this changing the property (position) of `buffer` compared to what it was (when the user invoked this method)? if so, maybe it's a good idea to document what would be the state of `buffer` once returned from the method, Or resetting it to the original state if that's the pattern (we can check the behavior of other platform APIs taking ByteBuffer )

Reply:
This is attempting to emulate what would previously happen when we were calling `ByteBuffer.get(byte[])`. Internally, `get` moves the position of the `ByteBuffer` for either the size of the `byte[]` or to the position being the limit of the `ByteBuffer`. So, this should be safe to do, and I'll call out that this method changes the ByteBuffer position in the future (as this comment is old and this moved into implementation)